07-task-aliasing.Rmd

```diff
@@ -0,0 +1,80 @@
+**6. Ensures Consistency**: Task aliasing assures that there is consistency in replicated tasks and helps the reader easily identify where changes are expected in a task. 
+
+To start aliasing your task the prerequisite is that the task must be defined first (which we have already done as a part of the previous sections). For this example we are going to alias the BwaMem task (and others as you will see in the example workflow) to run on the normal sample. 
+
+> Note: In this example workflow, purely for demonstration and learning purposes we have split the processing of the "tumor" samples and the "normal" sample. In the real world most paired samples would be processed together. For example you would want to pair the data from a single patient together. However you can leverage this task aliasing example for workflows where you may have multiple "test"/"tumor" samples and a single "normal" sample. 
+
+Here we are aliasing the task BwaMem to run this task for our "normal" samples. 
+
+First you will want to make sure that in your workflow input you are calling the sample you want to run this alias-ed task 
+
+```
+workflow mutation_calling {
+  input {
+    ...
+    File normal
+...
+  }
+
+Next all that you need to do is `call` the `task` you want to alias and use `as` to the `alias_of_your_choice`. 
+
+But don't forget to make sure that all the inputs reflect actually different things we want to run this task on.
+
+In this case we will be using a different sample and therefore the input_fastq is directed to the appropriate file source. 
+
+```
+  call BwaMem as normalBwaMem {
+    input:
+      input_fastq = normal,
+      refGenome = refGenome
+  }
+  ```
+
+ And finally you will also want to make sure that in your outputs section you are saving the appropriate outputs to reflect the task alias. 
+
+```
+output {
+File normalalignedBamSorted = normalBwaMem.analysisReadySorted
+}
+
+We can do this for the other two tasks in our workflow as well
+
+```
+call MarkDuplicates as normalMarkDuplicates {
+    input:
+      input_bam = normalBwaMem.analysisReadySorted
+  }
+
+  call ApplyBaseRecalibrator as normalApplyBaseRecalibrator {
+    input:
+      input_bam = normalMarkDuplicates.markDuplicates_bam,
+      input_bam_index = normalMarkDuplicates.markDuplicates_bai,
+      dbSNP_vcf = dbSNP_vcf,
+      dbSNP_vcf_index = dbSNP_vcf_index,
+      known_indels_sites_VCFs = known_indels_sites_VCFs,
+      known_indels_sites_indices = known_indels_sites_indices,
+      refGenome = refGenome
+  }
+  ```
+Now adding these steps to the workflow we will have our tumor and normal sample aligned and recalibrated and suitable for ingestion into the mutation calling step for a paired mutation calling using MuTect2. 
+
```

Comment:
Where's the entire workflow with task aliasing?

Reply:
Isnt that just the whole workflow then? I can add that